registry/api/protocol/eureka/instance_handler.go

```diff
@@ -108,14 +108,24 @@ func (routes *Routes) registerInstance(w rest.ResponseWriter, r *rest.Request) {
 		return
 	}
 
+	ext, err := buildExtensionFromInstance(inst)
+	if err != nil {
+		routes.logger.WithFields(log.Fields{
+			"namespace": r.Env[env.Namespace],
+			"error":     err,
+		}).Errorf("Failed to register instance %+v", inst)
+
```

Comment:
Need to `i18n.Error(r, w, http.StatusBadRequest, i18n.ErrorInstanceMetadataInvalid)` here.
Actually, in line 107, above, too.